### DISCUSSION TITLE
Using Addon "Component discovery"  - can't get the "Regular expression to match translation files against" right

### DISCUSSION THREAD

**Discussion by brianread108** [WeblateOrg/weblate#12021] [Q&A] [answered]
I am unable to get the format right for the translation files, I get this:
"Regular expression is missing named group "component", the simplest way to define it is (?P<component>.*)." when I input this:
"root/usr/share/smanager/lib/SrvMngr/I18N/Modules/{{ component|title }}/pofiles/{{ component }}_\*.po"

I want the component name to be the directory name (which is capitalised) or the file(s) name part which is the same as the directory, but all lower case.

**Comment by brianread108** [comment, marked as answer]
ok, after a lot more research and also asking chatGPT, I got this to work:

^root/usr/share/smanager/lib/SrvMngr/I18N/Modules/(?P\<component_cap\>[A-Z][a-zA-Z]*)/pofiles/(?P\<component\>[a-z]+)_(?P<language>[a-z]{2})\.po$

I've not come across the ?P\<name\> structure before. The docs could do with an example or two.

**Reply by nijel** [reply]
Eh, the documentation actually has a whole chapter with examples: https://docs.weblate.org/en/latest/admin/addons.html#component-discovery-examples

**Comment by brianread108** [comment]
Interesting - I seem to have (docker) code for v5.6.2, that docs link shows 5.7. I only installed last week direct from git following your docker instructions. Have I missed a release?

**Reply by nijel** [reply]
5.7 is currently being developed. I always post links to the latest doc version, but there was no change in this.
